core/src/main/java/io/temporal/samples/earlyreturn/EarlyReturn.java

```diff
@@ -0,0 +1,256 @@
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.samples.earlyreturn;
+
+import io.temporal.activity.ActivityInterface;
+import io.temporal.activity.ActivityMethod;
+import io.temporal.activity.ActivityOptions;
+import io.temporal.client.*;
+import io.temporal.failure.ApplicationFailure;
+import io.temporal.serviceclient.WorkflowServiceStubs;
+import io.temporal.worker.Worker;
+import io.temporal.worker.WorkerFactory;
+import io.temporal.workflow.*;
+import java.time.Duration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EarlyReturn {
+  private static final String TASK_QUEUE = "EarlyReturnTaskQueue";
+  private static final String UPDATE_NAME = "early-return";
+
+  public static void main(String[] args) {
+    WorkflowClient client = setupWorkflowClient();
+    startWorker(client);
+    runWorkflowWithUpdateWithStart(client);
+  }
+
+  private static WorkflowClient setupWorkflowClient() {
+    WorkflowServiceStubs service = WorkflowServiceStubs.newLocalServiceStubs();
+    return WorkflowClient.newInstance(service);
+  }
+
+  private static void startWorker(WorkflowClient client) {
+    WorkerFactory factory = WorkerFactory.newInstance(client);
+    Worker worker = factory.newWorker(TASK_QUEUE);
+
+    worker.registerWorkflowImplementationTypes(TransactionWorkflowImpl.class);
+    worker.registerActivitiesImplementations(new TransactionActivitiesImpl());
+
+    factory.start();
+    System.out.println("Worker started");
+  }
+
+  private static void runWorkflowWithUpdateWithStart(WorkflowClient client) {
+    Transaction tx =
+        new Transaction(
+            "Bob",
+            "Alice",
+            // Change this amount to a negative number to have initTransaction fail
+            10000);
+    WorkflowOptions options =
+        WorkflowOptions.newBuilder()
+            .setTaskQueue(TASK_QUEUE)
+            .setWorkflowId("early-return-workflow-" + System.currentTimeMillis())
+            .build();
+
+    WorkflowStub workflowStub = client.newUntypedWorkflowStub("TransactionWorkflow", options);
+
+    try {
+      System.out.println("Starting workflow with UpdateWithStart");
+
+      UpdateWithStartWorkflowOperation<String> update =
+          UpdateWithStartWorkflowOperation.newBuilder(UPDATE_NAME, String.class, new Object[] {})
+              .setWaitForStage(WorkflowUpdateStage.COMPLETED)
```

Comment:
Is there a reason you chose to use the untyped API here?

Reply:
Nope, will change to typed. Thanks

Reply:
This change is blocked by https://github.com/temporalio/sdk-java/issues/2285

Will update once it's fixed

Reply:
Tested with commit from fix PR https://github.com/temporalio/sdk-java/pull/2286